add new recipe to add gitignore entries

zieka opened the pull request.

timtebeek commented on rewrite-core/src/test/java/org/openrewrite/AddToGitignoreTest.java:
```diff
@@ -0,0 +1,258 @@
+            spec -> spec.path(".gitignore")
+          )
+        );
+    }
+}
```
It might be nice to avoid adding entries already matched by existing patterns.
```suggestion
    }

    @Test
    void doNotAddSuperfluousEntries() {
        rewriteRun(
          spec -> spec.recipe(new AddToGitignore("foo.tmp")),
          text(
            """
              *.tmp
              """,
            spec -> spec.path(".gitignore")
          )
        );
    }
}
```

zieka commented on rewrite-core/src/test/java/org/openrewrite/AddToGitignoreTest.java:
```diff
@@ -0,0 +1,258 @@
+        );
+    }
+
+    @Test
+    void handleNestedGitignore() {
+        rewriteRun(
+          spec -> spec.recipe(new AddToGitignore("local.properties")),
+          text(
+            """
+              *.class
+              """,
+            """
+              *.class
+              
+              local.properties
+              """,
+            spec -> spec.path("src/.gitignore")
+          )
+        );
+    }
```
good point

zieka commented:
> Let me know if you'd want to finish this yourself, or would rather hand over to me or someone on my team.

Hey just saw this; I think I addressed your comments.  I don't mind if someone from your team wants to continue on whatever would remain.
